onnxruntime/test/optimizer/qdq_transformer_test.cc

```diff
@@ -2140,5 +2142,113 @@ TEST(QDQTransformerTests, QDQ_Selector_Test) {
+    // If we enable cleanup and don't run the QDQ transformer we should drop all the Q->DQ pairs
+    auto build_test_case = [&](ModelTestBuilder& builder) {
+      auto input_count = input_shapes.size();
+      std::vector<NodeArg*> input_args;
+      std::vector<NodeArg*> q_input_args;
+      for (size_t i = 0; i < input_count; i++) {
+        input_args.push_back(builder.MakeInput<float>(input_shapes[i], -1.f, 1.f));
+        q_input_args.push_back(AddQDQNodePair<uint8_t>(builder, input_args.back(), 0.05f, 128));
+
+        if (i == 0 && block_removal_of_first_dq) {
+          // add another edge to the DQ node
+          auto* output = builder.MakeOutput();
+          builder.AddNode("Identity", {q_input_args.back()}, {output});
+        }
+      }
+      auto* concat_output = builder.MakeIntermediate();
+      Node& concat_node = builder.AddNode("Concat", q_input_args, {concat_output});
+      concat_node.AddAttribute("axis", int64_t(1));
+
+      auto* q_concat_output = builder.MakeIntermediate();
+      builder.AddQuantizeLinearNode<uint8_t>(concat_output, 0.05f, 128, q_concat_output);
+
+      auto* output_arg = builder.MakeOutput();
+      Node& dq_node = builder.AddDequantizeLinearNode<uint8_t>(q_concat_output, 0.05f, 128, output_arg);
+
+      if (block_removal_of_last_dq) {
+        // add another edge to the DQ node
+        auto* output = builder.MakeOutput();
+        builder.AddNode("Identity", {dq_node.MutableOutputDefs()[0]}, {output});
+      }
+    };
+
+    // if we block removal of the DQ node the Q node in the pair will not be removed either
+    int expected_qdq_count = 0 + (block_removal_of_first_dq ? 1 : 0) + (block_removal_of_last_dq ? 1 : 0);
+
+    auto check_graph = [expected_qdq_count](InferenceSessionWrapper& session) {
+      auto op_to_count = CountOpsInGraph(session.GetGraph());
+      EXPECT_EQ(op_to_count["QuantizeLinear"], expected_qdq_count);
+      EXPECT_EQ(op_to_count["DequantizeLinear"], expected_qdq_count);
+      EXPECT_EQ(op_to_count["Concat"], 1);
+    };
+
+    std::function<void(SessionOptions&)> func = [](SessionOptions& so) {
+      ASSERT_STATUS_OK(so.config_options.AddConfigEntry(kOrtSessionOptionsEnableQuantQDQCleanup, "1"));
+    };
+
+    // we increase the tolerance as removing the QDQ nodes means there's no round-trip to 8-bit and back
+    // essentially rounding the input values.
+    TransformerTester(build_test_case,
+                      check_graph,
+                      TransformerLevel::Level1,
+                      TransformerLevel::Level2,
+                      12 /*opset_version*/,
+                      0.025f /*per_sample_tolerance*/,
+                      0.01f /*relative_per_sample_tolerance*/,
+                      std::make_unique<QDQFinalCleanupTransformer>(),
+                      &func);
+  };
+
+  test_case({{1, 2, 4}, {1, 3, 4}});
+  // test_case({{1, 2, 4}, {1, 3, 4}}, true);         // block removal of first dq
```

Comment:
enable these tests too?

Reply:
Done. Was debugging a change and forgot to re-enable them.